go/ct/spc/specification_test.go

```diff
@@ -371,3 +374,71 @@ func TestSumWithOverflow(t *testing.T) {
 		})
 	}
 }
+
+func TestSpecification_OpsWithDynamicCostHandleOverflowSizes(t *testing.T) {
+
+	indexOfMemParam := map[vm.OpCode][]int{
+		vm.MCOPY:          {0, 1, 2},
+		vm.CODECOPY:       {0, 2},
+		vm.CALLDATACOPY:   {0, 2},
+		vm.RETURNDATACOPY: {0, 1, 2},
+		vm.REVERT:         {0, 1},
+		vm.CREATE:         {1, 2},
+		vm.CREATE2:        {1, 2},
+		vm.EXTCODECOPY:    {1, 3},
+		vm.CALLCODE:       {3, 4, 5, 6},
+		vm.SHA3:           {0, 1},
+		vm.STATICCALL:     {2, 3, 4, 5},
+	}
+
+	testValues := []uint64{st.MaxMemoryExpansionSize, st.MaxMemoryExpansionSize + 1, math.MaxUint64}
+	allRules := getAllRules()
+	rnd := rand.New(0)
+
+	for op := range indexOfMemParam {
+		for revision := common.MinRevision; revision <= common.NewestSupportedRevision; revision++ {
+			for _, overflowParameterPosition := range indexOfMemParam[op] {
+				for _, value := range testValues {
+					t.Run(fmt.Sprintf("%v_%v_%v_%v", op, revision, overflowParameterPosition, value), func(t *testing.T) {
+
+						condition := rlz.And(
+							rlz.IsRevision(revision),
+							rlz.Eq(rlz.Status(), st.Running),
+							rlz.Eq(rlz.Op(rlz.Pc()), vm.OpCode(op)),
+							rlz.Eq(rlz.Param(overflowParameterPosition), common.NewU256(value)),
+						)
+
+						for _, i := range indexOfMemParam[op] {
+							if i != overflowParameterPosition {
+								condition = rlz.And(condition, rlz.Eq(rlz.Param(i), common.NewU256(1)))
+							}
+						}
+
+						generator := gen.NewStateGenerator()
+						condition.Restrict(generator)
+
+						state, err := generator.Generate(rnd)
+						if err != nil {
+							t.Fatalf("failed to generate a random state: %v", err)
+						}
+
+						for _, rule := range allRules {
+							match, err := rule.Condition.Check(state)
+							if err != nil {
+								t.Fatalf("failed to check rule condition for %v: %v", rule.Name, err)
+							}
+							if match {
+								result := state.Clone()
+								rule.Effect.Apply(result)
+								if result.Status != st.Failed {
+									t.Fatalf("Rule %s fails with overflow parameters", rule.Name)
```

Comment:
Since you are checking that the status is not equal st.Failed the error message should say that it did NOT fail with overflow parameters. Or am I misunderstanding something?